### ISSUE TITLE
Leo input field should start as one line height and expand, allowing multiple lines

### ISSUE THREAD

**Issue by rebron** [brave/brave-browser#32062]
## Description 

Line height for the input field should start off as one line and expand.
And we should allow multiple lines (shift-enter or paste)

## Actual result:

<img width="382" alt="Screenshot 2023-08-02 at 1 26 49 PM" src="https://github.com/brave/brave-browser/assets/5951041/43e87792-4071-4dee-af64-15fe6512131d">

## Reproduces how often: 

Easily.

## Version/Channel Information:

- Can you reproduce this issue with the current release? n/a
- Can you reproduce this issue with the beta channel? n/a
- Can you reproduce this issue with the nightly channel? yes

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? n/a
- Does the issue resolve itself when disabling Brave Rewards? n/a
- Is the issue reproducible on the latest version of Chrome? n/a

**Comment by aguscruiz** [comment]
Adding screenshots if necessary:

Initial state
![image](https://github.com/brave/brave-browser/assets/15734630/e0636fe4-2c40-4eb5-84f5-6f82ee4a44f2)

Max height
![image](https://github.com/brave/brave-browser/assets/15734630/f4f5c329-362a-42c1-85d3-cf598bfadb0c)

**Comment by stephendonner** [comment]
## Verified `PASSED` using

```
Brave | 1.63.101 Chromium: 120.0.6099.199 (Official Build) nightly (x86_64)
-- | --
Revision | 8d3f0a563e185b6a763cc8e210ab24775dfb9047
OS | macOS Version 14.3 (Build 23D5043d)
```

### Steps:
1. install `1.63.101`
2. launched Brave
3. loaded `https://www.nbcnews.com/news/us-news/united-finds-bolts-need-tightening-inspections-door-plugs-grounded-boe-rcna132879`
4. clicked on `Leo` in the sidebar
5. clicked on `Accept and begin`
6. entered the following multi-line set of questions:
```
What's 4x4?
What's the square root of 16?
What's the square root of 100?
What's the current date and time?
What's the current weather?
```
7. clicked on the `Send message to Leo` button
8. waited for the response

### Confirmed `Leo` answered all `5` of my multi-line questions

<img width="1218" alt="Screenshot 2024-01-08 at 2 01 10 PM" src="https://github.com/brave/brave-browser/assets/387249/27060c62-2837-487d-8a79-f635ddee42ad">

**Comment by kjozwiak** [comment]
Removing `QA Pass` labels as the above will need to be re-checked using `1.62.x`. However, used https://github.com/brave/brave-browser/issues/32062#issuecomment-1881899077 as the needed verification to get the above uplifted into `1.62.x`.

Adding `QA/Blocked` for now as the above was merged into `cr121-1.62.x` via https://github.com/brave/brave-core/pull/21593. Once `C121` is merged into `1.62.x` via https://github.com/brave/brave-core/pull/21593, we'll unblock the above and mention the needed `1.62.x` version for verification 👍

**Comment by kjozwiak** [comment]
The above requires `1.62.148` or higher for `1.62.x` verification 👍

**Comment by stephendonner** [comment]
## Verified `PASSED` using

```
Brave | 1.62.149 Chromium: 121.0.6167.75 (Official Build) (x86_64)
-- | --
Revision | ff84587bd70af9fcbcbe59fc5194ca65082759c4
OS | macOS Version 11.7.10 (Build 20G1427)
```

### Steps:
1. installed `1.62.149`
2. launched Brave
3. opened a new-tab page
4. opened `Leo` in the sidebar
5. clicked on `Accept and begin`
6. pasted the following and clicked `return` to send to Leo:
```
What's 4x4?
What's the square root of 16?
What's the square root of 100?
What's the current date and time?
What's the current weather?
```
7. waited for `Leo` to answer

### Confirmed `Leo` answered my 5 questions

<img width="1312" alt="Screen Shot 2024-01-22 at 12 41 37 PM" src="https://github.com/brave/brave-browser/assets/387249/d0569429-68df-420f-94f8-09f7e2a7a23d">
